common/service/sensor/pldm_sensor.c

```diff
@@ -0,0 +1,353 @@
+			     int pldm_sensor_count, int thread_id, int sensor_num)
+{
+	CHECK_NULL_ARG(pldm_sensor_list);
+	CHECK_NULL_ARG(update_time);
+
+	int reading = 0;
+	uint8_t status = SENSOR_UNSPECIFIED_ERROR;
+
+	if (pldm_sensor_cfg->access_checker(sensor_num) != true) {
+		pldm_sensor_cfg->cache_status = PLDM_SENSOR_UNAVAILABLE;
+		return;
+	}
+
+	if (pldm_sensor_cfg->pre_sensor_read_hook) {
+		if (!pldm_sensor_cfg->pre_sensor_read_hook(pldm_sensor_cfg,
+							   pldm_sensor_cfg->pre_sensor_read_args)) {
+			pldm_sensor_cfg->cache_status = PLDM_SENSOR_FAILED;
+			*update_time = (k_uptime_get_32() / 1000);
+			LOG_DBG("Failed to pre read sensor_num 0x%x of thread %d", sensor_num,
+				thread_id);
+			return;
+		}
+	}
+
+	// TODO: Check device ready
+
+	if (pldm_sensor_cfg->read) {
+		status = pldm_sensor_cfg->read(pldm_sensor_cfg, &reading);
+
+		if ((status != SENSOR_READ_SUCCESS) && (status != SENSOR_READ_ACUR_SUCCESS)) {
+			pldm_sensor_cfg->cache_status = PLDM_SENSOR_FAILED;
+			*update_time = (k_uptime_get_32() / 1000);
+			LOG_ERR("Failed to read sensor_num 0x%x of thread %d", sensor_num,
+				thread_id);
+
+			// If sensor read fails, let the reading argument in the post_sensor_read_hook function to NULL.
+			// All post_sensor_read_hook function define in each platform should check reading
+			// whether is NULL to do the corresponding thing. (Ex: mutex_unlock)
+			if (pldm_sensor_cfg->post_sensor_read_hook) {
+				if (!pldm_sensor_cfg->post_sensor_read_hook(
+					    pldm_sensor_cfg, pldm_sensor_cfg->post_sensor_read_args,
+					    NULL)) {
+					LOG_DBG("Failed to pose read sensor_num 0x%x of thread %d",
+						sensor_num, thread_id);
+				}
+			}
+			return;
+		}
+	}
+
+	if (pldm_sensor_cfg->post_sensor_read_hook) {
+		if (!pldm_sensor_cfg->post_sensor_read_hook(
+			    pldm_sensor_cfg, pldm_sensor_cfg->post_sensor_read_args, &reading)) {
+			pldm_sensor_cfg->cache_status = PLDM_SENSOR_FAILED;
+			*update_time = (k_uptime_get_32() / 1000);
+			LOG_DBG("Failed to pose read sensor_num 0x%x of thread %d", sensor_num,
+				thread_id);
+			return;
+		}
+	}
+
+	pldm_sensor_cfg->cache_status = PLDM_SENSOR_ENABLED;
+	*update_time = (k_uptime_get_32() / 1000);
+	pldm_sensor_cfg->cache = reading;
+}
+
+void pldm_sensor_polling_handler(void *arug0, void *arug1, void *arug2)
+{
+	ARG_UNUSED(arug1);
+	ARG_UNUSED(arug2);
+
+	int ret = 0, sensor_num = 0;
+	int thread_id = (int)arug0;
+	int pldm_sensor_count = 0;
+
+	pldm_sensor_count = plat_pldm_sensor_get_sensor_count(thread_id);
+	if (pldm_sensor_count <= 0) {
+		LOG_ERR("Failed to get PLDM sensor count(%d) of thread%d", pldm_sensor_count,
+			thread_id);
+		return;
+	}
+
+	pldm_sensor_list[thread_id] = plat_pldm_sensor_load(thread_id);
+	if (pldm_sensor_list[thread_id] == NULL) {
+		LOG_ERR("Failed to load PLDM sensor list of thread%d, ret%d", thread_id, ret);
+		return;
+	}
+
+	for (sensor_num = 0; sensor_num < pldm_sensor_count; sensor_num++) {
```

Comment:
This is starting to get harder to read. Can you move the body if this loop into a function of its own?
That way the indentation is not all the way to the right and it gets harder to know where things start.
That way you can pass it &pldm_sensor_list[thread_id][sensor_num] and that function would be so much more easier to read.

Reply:
Modified with your suggestions. Thanks.